Documentation/Books/Manual/Programs/Arangorestore/ParallelRestore.md

```diff
@@ -1,2 +1,198 @@
+var files = fs.list(dumpDir).filter(f => !fs.isDirectory(f));
+var found = files.indexOf("ENCRYPTION");
+if (found === -1) {
+  print("This directory does not have an ENCRYPTION entry.");
+  exit(2);
+}
+// Remove ENCRYPTION entry:
+files = files.slice(0, found).concat(files.slice(found+1));
+
+for (let i = 0; i < files.length; ++i) {
+  if (files[i].slice(-5) !== ".json") {
+    print("This directory has files which do not end in '.json'!");
+    exit(3);
+  }
+}
+files = files.map(function(f) {
+  var fullName = fs.join(dumpDir, f);
+  var collName = "";
+  if (f.slice(-10) === ".data.json") {
+    var pos;
+    if (f.slice(0, 1) === "_") {  // system collection
+      pos = f.slice(1).indexOf("_") + 1;
+      collName = "_" + f.slice(1, pos);
+    } else {
+      pos = f.indexOf("_");
+      collName = f.slice(0, pos);
+    }
+  }
+  return {name: fullName, collName, size: fs.size(fullName)};
+});
+files = files.sort(function(a, b) { return b.size - a.size; });
+var dataFiles = [];
+for (let i = 0; i < files.length; ++i) {
+  if (files[i].name.slice(-10) === ".data.json") {
+    dataFiles.push(i);
+  }
+}
+
+// Produce the scripts, one for each coordinator:
+var scripts = [];
+for (let i = 0; i < coordinators.length; ++i) {
+  scripts.push([]);
+}
+
+var cnum = 0;
+for (let i = 0; i < dataFiles.length; ++i) {
+  var f = files[dataFiles[i]];
+  scripts[cnum].push(`${arangorestore} --collection ${f.collName} --input-directory ${dumpDir} --server.endpoint ${coordinators[cnum]} ` + otherArgs.join(" "));
+  cnum += 1;
+  if (cnum >= coordinators.length) {
+    cnum = 0;
+  }
+}
+
+for (let i = 0; i < coordinators.length; ++i) {
+  let f = "coordinator_" + i + ".sh";
+  print("Writing file", f, "...");
+  fs.writeFileSync(f, scripts[i].join("\n"));
+}
+EOF
+
+${ARANGOSH} --javascript.execute /tmp/parallelRestore$$.js -- "$@"
+rm /tmp/parallelRestore$$.js
+```
+
+To run this script, all coordinator endpoints of the cluster have to be
+provided. The example below is a template for a cluster with 3 coordinators:
+
+```
+./parallelRestore.js <dump-directory>
+  tcp://<ip-of-coordinator1>:<port of coordinator1>,
+  tcp://<ip-of-coordinator2>:<port of coordinator2>,
+  tcp://<ip-of-coordinator3>:<port of coordinator3>
+  --server.username <username>
+  --server.password <password>
+  --server.database <database_name>
+  --create-collection false
+```
+
+This time the option `--create-collection false` is used since we already
+created the document structure before. The above command will create 3 scripts,
+whereas 3 corresponds to the amount of listed coordinators. The resulting
+scripts are named _coordinator_number-of-coordinator.sh (e.g.
+coordinator_0.sh, coordinator_1.sh, coordinator_2.sh).
+
+### Execute parallel restore scripts
+
+The _coordinator_number-of-coordinator.sh scripts, that were created in the
+previous script, now have to be executed on each machine where a coordinator
+is running. This will start a parallel restore of the `dump`.
```

Comment:
"dump" does not refer to the folder here, why the code tags?

Reply:
removed code tags